service/matching/user_data_manager.go

```diff
@@ -337,14 +337,23 @@ func (m *userDataManagerImpl) fetchUserData(ctx context.Context) error {
 
 	for ctx.Err() == nil {
 		start := time.Now()
+		m.lock.Lock()
+		previousVersion := m.userData.GetVersion()
+		m.lock.Unlock()
+
 		_ = backoff.ThrottleRetryContext(ctx, op, m.config.GetUserDataRetryPolicy, nil)
+
+		m.lock.Lock()
+		currentVersion := m.userData.GetVersion()
+		m.lock.Unlock()
```

Comment:
Instead of these GetVersion calls, can we just have `op` set a local variable to false at the top of the function, and set it to true when it gets a new version? Then we can just use that value below